### PR TITLE
Engine stats

### PR DESCRIPTION
**Pull request by chrisnojima**
- [ ] ~add to log send~ (turns out this isn't easy to plumb right now)
- [x] whitelist for kb

### CODE REVIEW

File: shared/engine/transport-shared.js [hunk 1]
```diff
@@ -7,19 +7,13 @@ import {requestIdleCallback} from '../util/idle-callback'
...
 import type {rpcLogType} from './index.platform'
 
-const {
-  transport: {RobustTransport},
-  client: {Client: RpcClient},
-} = rpc
-const KEYBASE_RPC_DELAY_RESULT: number = process.env.KEYBASE_RPC_DELAY_RESULT
-  ? parseInt(process.env.KEYBASE_RPC_DELAY_RESULT)
-  : 0
-const KEYBASE_RPC_DELAY: number = process.env.KEYBASE_RPC_DELAY ? parseInt(process.env.KEYBASE_RPC_DELAY) : 0
```
Comment (chrisnojima) [inline shared/engine/transport-shared.js]: i'm sure no one uses this so lets just remove it

File: shared/actions/tracker.js [hunk 2]
```diff
@@ -618,11 +618,11 @@ function _setupTrackerHandlers() {
         }
       }
 
-      const session: Session = engine().createSession(
-        _serverCallMap(dispatch, getState, onStart, onFinish),
-        null,
-        cancelHandler
-      )
+      const session: Session = engine().createSession({
```
Comment (chrisnojima) [inline shared/actions/tracker.js]: lets use named parameters so its less confusing

File: shared/actions/unlock-folders.js [hunk 3]
```diff
@@ -71,10 +72,8 @@ function _registerRekeyListener() {
         },
         'keybase.1.rekeyUI.rekeySendEvent': () => {}, // ignored debug call from daemon
       },
-      null,
-      null,
-      true
-    )
+      startMethod: 'keybase.1.rekeyUI.delegateRekeyUI',
```
Comment (chrisnojima) [inline shared/actions/unlock-folders.js]: we now add a hint about the method even though its not required, helps logging

File: shared/engine/transport-shared.js [hunk 4]
```diff
@@ -2,113 +2,90 @@
 // Classes used to handle RPCs. Ability to inject delays into calls to/from server
 import rpc from 'framed-msgpack-rpc'
 import {localLog} from '../util/forward-logs'
```
Comment (chrisnojima) [inline shared/engine/transport-shared.js]: generally cleaning up this file cause its pretty old

File: shared/engine/transport-shared.js [hunk 5]
```diff
@@ -2,113 +2,90 @@
...
-import {printRPC} from '../local-debug'
+import {printRPC, printRPCStats} from '../local-debug'
 import {requestIdleCallback} from '../util/idle-callback'
-
-import type {rpcLogType} from './index.platform'
-
-const {
-  transport: {RobustTransport},
-  client: {Client: RpcClient},
-} = rpc
```
Comment (chrisnojima) [inline shared/engine/transport-shared.js]: I find this kind of deeply nested destructuring harder to read

File: shared/engine/transport-shared.js [hunk 6]
```diff
@@ -2,113 +2,90 @@
...
+  methodInCallback: boolean,
+}): F {
+  const {handler, extra, method, type, enforceOnlyOnce, methodInCallback} = options
   let once = false
-  // $FlowIssue
-  return (...args) => {
-    if (once) {
-      rpcLog('engineInternal', 'ignoring multiple result calls', {args})
+  // $ForceType
+  const wrapped: F = (a1: A1, a2: A2, a3: A3, a4: A4, a5: A5): void => {
```
Comment (chrisnojima) [inline shared/engine/transport-shared.js]: having one wrapping function is a lot simpler to understand than all these nested calls that make wrappers

File: shared/engine/transport-shared.js [hunk 7]
```diff
@@ -2,113 +2,90 @@
...
+    connectCallback: () => void,
+    disconnectCallback: () => void,
+    incomingRPCCallback: (a: any) => void,
+    writeCallback: any
+  ) {
+    super(opts)
+
+    this.hooks = {
       connected: () => {
-        // $FlowIssue complains that this might be null
```
Comment (chrisnojima) [inline shared/engine/transport-shared.js]: flow rightly complains about using this before super is called. turns out we can just set hooks after

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready. This adds a little black bar for us in the app (even in prod mode). you can click it to hide it. other users can turn this on if they turn on the flag

**Comment by chrisnojima** [comment]
@keybase/react-hackers bump

**Comment by buoyad** [comment]
Oops meant to comment

**Pull request merged**